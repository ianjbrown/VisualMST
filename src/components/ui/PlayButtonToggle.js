import PlayArrowIcon from "@mui/icons-material/PlayArrow";
import PauseOutlinedIcon from "@mui/icons-material/PauseOutlined";

function PlayButtonToggle(props) {

  function clickHandler() {
    props.onClick();
  }
  
  if (props.paused) {
    return (
      <PlayArrowIcon
        sx={{ "&:hover": { cursor: "pointer" }, fontSize: 60 }}
        onClick={clickHandler}
      />
    );
  } else {
    return (
      <PauseOutlinedIcon
        sx={{ "&:hover": { cursor: "pointer" }, fontSize: 60 }}
        onClick={clickHandler}
      />
    );
  }
}

export default PlayButtonToggle;
