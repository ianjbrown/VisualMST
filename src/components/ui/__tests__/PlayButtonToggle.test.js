import PlayButtonToggle from "../PlayButtonToggle";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Play matches snapshot", () => {
    const play = renderer.create(<PlayButtonToggle paused={true} />);
    expect(play).toMatchSnapshot();
})

test("Pause matches snapshot", () => {
    const pause = renderer.create(<PlayButtonToggle paused={false} />);
    expect(pause).toMatchSnapshot();
})

test("Correct button displayed when paused", () => {
    render(<PlayButtonToggle paused={true} />);
    const playButton = screen.queryByTestId("PlayArrowIcon");
    const pauseButton = screen.queryByTestId("PauseOutlinedIcon");
    expect(playButton).toBeInTheDocument();
    expect(pauseButton).not.toBeInTheDocument();
})

test("Correct button displayed when not paused", () => {
    render(<PlayButtonToggle paused={false} />);
    const playButton = screen.queryByTestId("PlayArrowIcon");
    const pauseButton = screen.queryByTestId("PauseOutlinedIcon");
    expect(playButton).not.toBeInTheDocument();
    expect(pauseButton).toBeInTheDocument();


})