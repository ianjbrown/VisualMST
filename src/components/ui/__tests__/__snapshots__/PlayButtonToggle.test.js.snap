// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pause matches snapshot 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
  data-testid="PauseOutlinedIcon"
  focusable="false"
  onClick={[Function]}
  viewBox="0 0 24 24"
>
  <path
    d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
  />
</svg>
`;

exports[`Play matches snapshot 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tvxjov-MuiSvgIcon-root"
  data-testid="PlayArrowIcon"
  focusable="false"
  onClick={[Function]}
  viewBox="0 0 24 24"
>
  <path
    d="M8 5v14l11-7z"
  />
</svg>
`;
